src/compojure/api/middleware.clj

```diff
@@ -129,19 +133,10 @@
 
 (defn ->mime-types [formats] (map mime-types formats))
 
-(defn handle-req-error [^Throwable e handler req]
-  (cond
-    (instance? JsonParseException e)
-    (bad-request {:type "json-parse-exception"
-                  :message (.getMessage e)})
-
-    (instance? ParserException e)
-    (bad-request {:type "yaml-parse-exception"
-                  :message (.getMessage e)})
-
-    :else
-    (internal-server-error {:type (str (class e))
-                            :message (.getMessage e)})))
+(defn handle-req-error [^Throwable e handler request]
+  (if (or (instance? JsonParseException e) (instance? ParserException e))
+    (throw+ {:type ::ex/request-validation} e)
```

Comment:
New type means also new handler config to override for users.
Anyhow now the original exception type and message is missed